#' Retrieve MutSig final analysis MAF.
#'
#' This service returns selected columns from the MAF generated by MutSig.
#' Results may be filtered by gene, cohort, tool, or barcode, but at least one
#' gene OR barcode OR cohort must be given.
#'
#' \code{MutSig2.0}, \code{MutSig2CV} (default) and \code{MutSigCV}. For more
#' details please visit
#' \url{https://www.broadinstitute.org/cancer/cga/mutsig_faq}.
#' @param column Character vector which data columns/fields to return. Every
#' subset of \code{colnames} is possible.
#' @inheritParams Samples.mRNASeq
#' @inheritParams Archives.StandardData

#' @details
#' Not specifying a patient-, gene- or cohort subset will return all Variants.
#' This can result in a lot of pages.
#'
#' @examples
#' format = "json"
#' cohort = ""
#' tool = "MutSig2CV"
#' gene = c("TP53", "RUNX1")
#' tcga_participant_barcode = "TCGA-CH-5761"
#' column = ""
#' page = 1
#' page_size = 250
#' sort_by = c("gene")
#'
#' obj = Analyses.Mutation.MAF(format = format,
#'                                 cohort = cohort,
#'                                 tool = tool,
#'                                 gene =gene,
#'                                 tcga_participant_barcode = tcga_participant_barcode,
#'                                 column = column,
#'                                 page = page,
#'                                 page_size = page_size,
#'                                 sort_by = sort_by)
#'
#' format = "csv"
#' obj = Analyses.Mutation.MAF(format = format,
#'                                 cohort = cohort,
#'                                 tool = tool,
#'                                 gene =gene,
#'                                 tcga_participant_barcode = tcga_participant_barcode,
#'                                 column = column,
#'                                 page = page,
#'                                 page_size = page_size,
#'                                 sort_by = sort_by)
#'
#' gene = c("TP53", "RUNX1")
#' tcga_participant_barcode = "TCGA-CH-5761"
#' cohort = "PRAD"
#' obj = Analyses.Mutation.MAF(format = format,
#'                                 cohort = cohort,
#'                                 tool = tool,
#'                                 gene =gene,
#'                                 tcga_participant_barcode = tcga_participant_barcode,
#'                                 column = column,
#'                                 page = page,
#'                                 page_size = page_size,
#'                                 sort_by = sort_by)
#'
#' gene = ""
#' obj = Analyses.Mutation.MAF(format = format,
#'                                 cohort = cohort,
#'                                 tool = tool,
#'                                 gene = gene,
#'                                 tcga_participant_barcode = tcga_participant_barcode,
#'                                 column = column,
#'                                 page = page,
#'                                 page_size = page_size,
#'                                 sort_by = sort_by)
#'
#' column = c("tcga_participant_barcode", "Hugo_Symbol", "Variant_Type")
#' obj = Analyses.Mutation.MAF(format = format,
#'                                 cohort = cohort,
#'                                 tool = tool,
#'                                 gene = gene,
#'                                 tcga_participant_barcode = tcga_participant_barcode,
#'                                 column = column,
#'                                 page = page,
#'                                 page_size = page_size,
#'                                 sort_by = sort_by)
#'
#' @return A \code{list}, if format is \code{json}, elsewise a \code{data.frame}
#'
#' @export
Analyses.Mutation.MAF = function(format = "csv",
                               cohort = "",
                               tool = "MutSig2CV",
                               gene = "",
                               tcga_participant_barcode = "",
                               column = "",
                               page = 1,
                               page_size = 250,
                               sort_by = "gene"){

  parameters = list(format = format,
                    cohort = cohort,
                    tool = tool,
                    gene =gene,
                    tcga_participant_barcode = tcga_participant_barcode,
                    column = column,
                    page = page,
                    page_size = page_size,
                    sort_by = sort_by)

  to.Validate = c("gene", "cohort", "tcga_participant_barcode")
  validet.Parameters(params = parameters, to.Validate = to.Validate)
  url = build.Query(parameters = parameters, invoker = "Analyses", method = "Mutation/MAF")

  ret = download.Data(url, format, page)

  return(ret)
}
